include/clang/AST/Stmt.h

```diff
@@ -252,18 +252,11 @@ class LLVM_ALIGNAS(LLVM_PTR_SIZE) Stmt {
     unsigned NumArgs : 32 - 8 - 1 - NumExprBits;
   };
 
-  class CountBoundsExprBitFields {
-    friend class CountBoundsExpr;
+  class BoundsExprBitFields {
+    friend class BoundsExpr;
 
     unsigned : NumExprBits;
-    unsigned Kind : 1;
-  };
-
-  class NullaryBoundsExprBitFields {
-    friend class NullaryBoundsExpr;
-
-    unsigned : NumExprBits;
-    unsigned Kind : 1;
+    unsigned Kind : 3;
```

Comment:
clang code generally doesn't safe guard against this kind of mistake, although it could, as this is a somewhat difficult to debug mistake if you make it.

I've added a constant that names the size of the bit field. and added a static assert to guard against this.   I used the trick of using an enum member to define the constant.